firebase/functions/src/uploads/add-outputs.ts

```diff
@@ -1,20 +1,22 @@
 import * as admin from 'firebase-admin';
-import * as functions from 'firebase-functions';
-import { TriggerAnnotated, Event } from 'firebase-functions';
+import { storage } from 'firebase-functions';
+
 import { ObjectMetadata } from 'firebase-functions/lib/providers/storage';
+import { Runnable, TriggerAnnotated } from 'firebase-functions/lib/cloud-functions';
+
 import { UploadType } from './upload-type';
 
-export const bucketChange = functions.storage.object().onChange(event => {
+export const bucketChange = storage.object().onFinalize((data, context) => {
   if (
-    !event.data ||
-    event.data.metageneration > 1 ||
-    !event.data.metadata ||
-    event.data.metadata['type'] !== UploadType.ExecutionOutput
+    !data ||
+    parseInt(data.metageneration, 10) > 1 ||
```

Comment:
Does that mean they changed that from `Number` (previously) to `String` (now)? Or was it always `String` and we just didn't properly handle it?

Reply:
It was in fact always a string (v0.9.x https://firebase.google.com/docs/reference/functions/beta/functions.storage.ObjectMetadata#metageneration vs. v1.x https://firebase.google.com/docs/reference/functions/functions.storage.ObjectMetadata#metageneration)